pkg/dbutils/postgresql_utils.go

```diff
@@ -171,22 +170,83 @@ CREATE TABLE IF NOT EXISTS %s (
 	UNIQUE(repo_namespace, chart_files_ID),
 	FOREIGN KEY (repo_name, repo_namespace) REFERENCES %s (name, namespace) ON DELETE CASCADE,
 	FOREIGN KEY (repo_name, repo_namespace, chart_id) REFERENCES %s (repo_name, repo_namespace, chart_id) ON DELETE CASCADE
-)`, ChartFilesTable, RepositoryTable, ChartTable))
+)`, filesTable, RepositoryTable, ChartTable))
+	return err
+}
+
+// InitTables creates the required tables for the postgresql backend for assets.
+func (m *PostgresAssetManager) InitTables() error {
+	// Repository table should have a namespace column, and chart table should reference repositories.
+	err := m.createRepositoryTable(RepositoryTable)
 	if err != nil {
-		return err
+		return fmt.Errorf("Unable to create %s table: %v", RepositoryTable, err)
+	}
+
+	err = m.createChartTable(ChartTable)
+	if err != nil {
+		return fmt.Errorf("Unable to create %s table: %v", ChartTable, err)
+	}
+
+	err = m.createFilesTable(ChartFilesTable)
+	if err != nil {
+		return fmt.Errorf("Unable to create %s table: %v", ChartFilesTable, err)
 	}
 	return nil
 }
 
+func (m *PostgresAssetManager) tablesAreEqual(table1, table2 string) bool {
+	_, err := m.DB.Exec(fmt.Sprintf(`select * from (
+    (select * from %s limit 0) 
+    union all
+    (select * from %s limit 0)
+) as test`, table1, table2))
+	// If the two tables can be joined, they are compatible
```

Comment:
> On startup, within a transaction the binary checks for the value of the schema version from the db (if it exists) and if it doesn't match, recreates the schema. This is the kind of simple schema handling we should have included when first adding the postgres support, imo.

_Needs to read more about transactions_

> we could even calculate a hash and store that rather than a version, to ensure devs can't change the schema without causing a change

Yes, I was going to suggest that, it's safer to do it that way.